samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -16,85 +16,378 @@
+   *
+   * <pre> {@code
+   * MessageStream<Long> stream = ...;
+   * BiFunction<Long, Long, Long> maxAggregator = (m, c)-> Math.max(m, c);
+   * MessageStream<WindowOutput<WindowKey, Long>> windowed = stream.window(Windows.globalWindow(maxAggregator)
+   * .setEarlyTriggers(Triggers.repeat(Triggers.any(Triggers.count(50), Triggers.timeSinceFirstMessage(Duration.seconds(10))))))
+   * }
+   * </pre>
+   *
+   * @param aggregateFunction the function to aggregate window results
+   * @param <M> the type of {@link MessageEnvelope}
+   * @param <WV> type of the {@link WindowOutput} output value
+   * @return the created {@link WindowFunction} function.
    */
-  public static <M extends MessageEnvelope, WK, SI> Window<M, WK, Collection<SI>, WindowOutput<WK, Collection<SI>>> intoSessions(Function<M, WK> sessionKeyFunction,
-      Function<M, SI> sessionInfoExtractor) {
-    return new SessionWindow<>(sessionKeyFunction, (m, c) -> {
-        c.add(sessionInfoExtractor.apply(m));
-        return c;
-      }
-    );
+  public static <M extends MessageEnvelope, WV> WindowFunction<M, Void, WindowKey<Void>, WV, WindowOutput<WindowKey<Void>, WV>> globalWindow(BiFunction<M, WV, WV> aggregateFunction) {
+    return new BaseWindowFunction<M, Void, WV>(null, aggregateFunction, null, null);
   }
 
   /**
-   * Static API method to create a {@link SessionWindow} as a counter of input {@link MessageEnvelope}s
+   * Returns a {@link WindowFunction} that windows all values into a single global window. This window does not have a
+   * default trigger. The triggering behavior must be specified by setting early triggers using
+   * {@link org.apache.samza.operators.triggers.Triggers} APIs.
+   *
+   * For example, to window a stream into count based windows that trigger every 50 messages or every 10 minutes.
+   * <pre> {@code
+   * MessageStream<Long> stream = ...;
+   * MessageStream<WindowOutput<WindowKey, Collection<Long>>> windowed = stream.window(Windows.globalWindow()
+   * .setEarlyTrigger(Triggers.repeat(Triggers.any(Triggers.count(50), Triggers.timeSinceFirstMessage(Duration.seconds(10))))))
+   * }
+   * </pre>
    *
-   * @param sessionKeyFunction  function to calculate session window key
-   * @param <M>  type of the input {@link MessageEnvelope}
-   * @param <WK>  type of the session window key
-   * @return  the {@link Window} function for the session
+   * @param <M> the type of {@link MessageEnvelope}
+   * @return the created {@link WindowFunction} function.
    */
-  public static <M extends MessageEnvelope, WK> Window<M, WK, Integer, WindowOutput<WK, Integer>> intoSessionCounter(Function<M, WK> sessionKeyFunction) {
-    return new SessionWindow<>(sessionKeyFunction, (m, c) -> c + 1);
+
+  public static <M extends MessageEnvelope> WindowFunction<M, Void, WindowKey<Void>, Collection<M>, WindowOutput<WindowKey<Void>, Collection<M>>> globalWindow() {
```

Comment:
Same question here: no start and end, means only one continuous, never ending pane?